operators/cmd/tenant-operator/main.go

```diff
@@ -131,36 +133,44 @@ func main() {
 		log.Error(err, "Unable to create manager")
 		os.Exit(1)
 	}
+	if enableWH {
 
-	if err = (&clv1alpha1.Tenant{}).SetupWebhookWithManager(mgr); err != nil {
-		log.Error(err, "unable to create conversion webhook", "webhook", "Tenant")
-		os.Exit(1)
-	}
-	if err = (&clv1alpha2.Tenant{}).SetupWebhookWithManager(mgr); err != nil {
-		log.Error(err, "unable to create conversion webhook", "webhook", "Tenant")
-		os.Exit(1)
+		if err = (&clv1alpha1.Tenant{}).SetupWebhookWithManager(mgr); err != nil {
+			log.Error(err, "unable to create conversion webhook", "webhook", "Tenant")
+			os.Exit(1)
+		}
+		if err = (&clv1alpha2.Tenant{}).SetupWebhookWithManager(mgr); err != nil {
+			log.Error(err, "unable to create conversion webhook", "webhook", "Tenant")
+			os.Exit(1)
+		}
+
+		hookServer := mgr.GetWebhookServer()
+		webhookBypassGroupsList := strings.Split(webhookBypassGroups, ",")
+		hookServer.Register(ValidatingWebhookPath, tenantwh.MakeTenantValidator(mgr.GetClient(), webhookBypassGroupsList))
+		hookServer.Register(MutatingWebhookPath, tenantwh.MakeTenantLabeler(mgr.GetClient(), webhookBypassGroupsList, targetLabelKey, targetLabelValue))
 	}
 
-	hookServer := mgr.GetWebhookServer()
-	webhookBypassGroupsList := strings.Split(webhookBypassGroups, ",")
-	hookServer.Register(ValidatingWebhookPath, tenantwh.MakeTenantValidator(mgr.GetClient(), webhookBypassGroupsList))
-	hookServer.Register(MutatingWebhookPath, tenantwh.MakeTenantLabeler(mgr.GetClient(), webhookBypassGroupsList, targetLabelKey, targetLabelValue))
+	var kcA *controllers.KcActor
+	if kcURL != "DISABLE" {
```

Comment:
Would personally skip the creation if the URL is empty, instead of using custom values (should adapt the parameters check above). Also in this case add a warning level log message if skipped. Same for keycloak

Reply:
Since we have to set an empty string instead of "DISABLE", do we have to remove the checks `kcURL == ""` and `ncURL == ""` at line 107?

Reply:
I'd say yes

Reply:
Should also split the other checks and move them to the appropriate location.